File: RegisteredUser.java
import java.util.ArrayList;

public class RegisteredUser extends User{
    private int points;
    private String paymentMethod;
    private ArrayList<PurchaseOrder> purchaseOrders;
    private ArrayList<ReturnOrder> returnOrders;

    //default ctor
    public RegisteredUser(){
        super();
        this.points = 0;
        this.purchaseOrders = new ArrayList<PurchaseOrder>();
        this.returnOrders = new ArrayList<ReturnOrder>();
    }

    //ctor
    public RegisteredUser(int id, String n, String a, String un, String pw, String e, String bd, int p, String pm, 
        ArrayList<PurchaseOrder> po, ArrayList<ReturnOrder> ro){
        super(id, n, a, un, pw, e, bd);
        this.points = p;
        this.paymentMethod = pm;
        this.purchaseOrders = po;
        this.returnOrders = ro;
    }

    //getters
    public int getPoints(){ return this.points; }
    public String getPaymentMethods(){ return this.paymentMethod; }
    public ArrayList<PurchaseOrder> getPurchaseOrders(){ return this.purchaseOrders; }
    public ArrayList<ReturnOrder> getReturnOrders(){ return this.returnOrders; }
    //add methods to find orders based on id?

    //setters
    public void setPoints(int p){ this.points = p; }
    public void setPaymentMethods(String pm){ this.paymentMethod = pm; }
    public void setPurchaseOrders(ArrayList<PurchaseOrder> po){ this.purchaseOrders = po; }
    public void setReturnOrders(ArrayList<ReturnOrder> ro){ this.returnOrders = ro; }

    public void addPoints(int p){
        this.points += p;
    }

    public void removePoints(int p){
        this.points -= p;
    }

    public void addPurchase(PurchaseOrder p){
        this.purchaseOrders.add(p);
    }

    public void addReturn(ReturnOrder r){
        this.returnOrders.add(r);
    }
}

File: model/RegisteredUser.php
<?php

class RegisteredUser extends User{
    private $points = 0;
    private $paymentMethod;
    private $purchaseOrders = array();
    private $returnOrders = array();

    //ctor
    public function __construct(&$id, &$n, &$a, &$un, &$pw, &$e, &$bd, &$p, &$pm,
                                &$po, &$ro){
        parent::__construct($id, $n, $a, $un, $pw, $e, $bd);
        $this->points = $p;
        $this->paymentMethod = $pm;
        $this->purchaseOrders = $po;
        $this->returnOrders = $ro;
    }

    //getters
    public function getPoints(){ return $this->points; }
    public function getPaymentMethods(){ return $this->paymentMethod; }
    public function getPurchaseOrders(){ return $this->purchaseOrders; }
    public function getReturnOrders(){ return $this->returnOrders; }
    //add methods to find orders based on id?

    //setters
    public function setPoints(&$p){ $this->points = $p; }
    public function setPaymentMethods(&$pm){ $this->paymentMethod = $pm; }
    public function setPurchaseOrders(&$po){ $this->purchaseOrders = $po; }
    public function setReturnOrders(&$ro){ $this->returnOrders = $ro; }

    public function addPoints(&$p){
        $this->points += $p;
    }

    public function removePoints(&$p){
        $this->points -= $p;
    }

    public function addPurchase(&$p){
        array_push($this->purchaseOrders, $p);
    }

    public function addReturn(&$r){
        array_push($this->returnOrders, $r);
    }
}

?>
